File: 588a/main.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)

	a := make([]int, n)
	p := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i], &p[i])
	}

	buy := a[0]
	price := p[0]
	cost := 0
	for i := 1; i < n; i++ {
		if p[i] >= price {
			buy += a[i]
		} else {
			cost += buy * price
			buy = a[i]
			price = p[i]
		}
	}

	cost += buy * price
	fmt.Print(cost)
}

File: 588a/main.c
#include <stdlib.h>
#include <stdio.h>

int main() {
    int n, q, p;
    scanf("%d", &n);

    int buy, price;
    scanf("%d %d", &buy, &price);

    int i = 1;
    long long cost = 0;
    for (; i < n; i++) {
        scanf("%d %d", &q, &p);
        if (p >= price) {
            buy += q;
        } else {
            cost += (long long)buy * price;
            buy = q;
            price = p;
        }
    }

    cost += (long long)buy * price;
    printf("%d", cost);
    

}
